domains/certificates/Query.API/API/DataSyncSyncer/DataSyncService.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Security.Claims;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using API.DataSyncSyncer.Dto;
+using CertificateEvents.Primitives;
+using Marten;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.DataSyncSyncer;
+
+public class DataSyncService
```

Comment:
I see this class as having multiple responsibilities: 

1) Updating the sync state (periodStartTimeDictionary)
2) Logic about a creating a period to request measurements from and if the period is valid
3) Calling DataSync

Given that we will add storage for persisting the sync state, then it would make sense to extract a class/interface for this storage. This will also remove some of the responsibility from this class